deployment/windows/jellyfin.nsi

```diff
@@ -0,0 +1,202 @@
+; Shows a lot of debug information while compiling
+; This can be removed once stable.
+!verbose 4
+;--------------------------------
+;Include Modern UI
+
+    !include "MUI2.nsh"
+    Var _JELLYFINVERSION_
+;    Var _DEFAULTEMBYDATADIR_
+    Var _JELLYFINDATADIR_
+    Var _SERVICEINSTALLED_
+;--------------------------------
+;General
+
+; Align installer version with jellyfin.dll version
+    !getdllversion "$%InstallLocation%\jellyfin.dll" ver_
+    !echo "jellyfin.dll version is ${ver_1}.${ver_2}.${ver_3}.${ver_4}" ;!echo will print it while building
+
+    Name "Jellyfin ${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    OutFile "Jellyfin.Installer.${ver_1}.${ver_2}.${ver_3}.${ver_4}.exe"
+    BrandingText "Jellyfin ${ver_1}.${ver_2}.${ver_3}.${ver_4} Installer"
+
+; installer attributes
+    VIProductVersion "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    VIFileVersion "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    VIAddVersionKey "ProductName" "Jellyfin"
+    VIAddVersionKey "FileVersion" "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+  
+  ;Default installation folder
+    InstallDir "$APPDATA\Jellyfin"
+  
+  ;Get installation folder from registry if available
+    InstallDirRegKey HKLM "Software\Jellyfin" "InstallLocation"
+  
+  ;Request application privileges for service installation
+    RequestExecutionLevel admin
+    CRCCheck on	    
+    !define MUI_ABORTWARNING
+
+;--------------------------------
+;Pages
+
+    !insertmacro MUI_PAGE_LICENSE "$%InstallLocation%\LICENSE"
+    !insertmacro MUI_PAGE_COMPONENTS
+    !insertmacro MUI_PAGE_DIRECTORY
+
+; Custom Directory page to ask for Emby Library location in case its needed
+; Commented for now to avoid showing this. 
+; This can be uncommented in case Emby Migration is planned later
+;    !define MUI_PAGE_HEADER_TEXT "Emby Library locaton"
+;    !define MUI_PAGE_HEADER_SUBTEXT  ""
+;    !define MUI_DIRECTORYPAGE_TEXT_TOP "Please select the folder where Emby library is present. This will have Enby folders like config, cache, data, metadata, etc."
+;    !define MUI_DIRECTORYPAGE_TEXT_DESTINATION "Emby Library location"
+;    !define MUI_PAGE_CUSTOMFUNCTION_PRE ShowEmbyLibraryPage
+;    !define MUI_DIRECTORYPAGE_VARIABLE $_DEFAULTEMBYDATADIR_
+;    !insertmacro MUI_PAGE_DIRECTORY
+	
+    !insertmacro MUI_PAGE_INSTFILES
+
+    !insertmacro MUI_UNPAGE_CONFIRM
+    !insertmacro MUI_UNPAGE_INSTFILES
+  
+;--------------------------------
+;Languages
+ 
+    !insertmacro MUI_LANGUAGE "English"
+
+;--------------------------------
+;Installer Sections
+
+Section "Install Jellyfin (required)" InstallJellyfin
+    SetOutPath "$INSTDIR"
+    SetShellVarContext current
+    StrCpy $_JELLYFINDATADIR_ "$LOCALAPPDATA\jellyfin\"
+
+; Pack all the files that were just compiled
+    File /r $%InstallLocation%\* 
+	
+; Write the installation path into the registry
+    WriteRegStr HKLM "Software\Jellyfin" "InstallLocation" "$INSTDIR"
```

Comment:
This is (at the moment) fine. In general it would be best to avoid the registry at all, but since it is a lowkey req for the clean installer/uninstaller in a non UWP context this is as good a place as any to put it. We should document that we write the install location to there for people obsessive about clean registries.

Reply:
Agree.